src/ncp/ncp_base.cpp

```diff
@@ -414,8 +416,8 @@ void NcpBase::HandleSendDone()
     if (mSendQueue.GetHead() != NULL)
     {
         Message &message(*mSendQueue.GetHead());
-        HandleDatagramFromStack(message);
         mSendQueue.Dequeue(message);
+        HandleDatagramFromStack(message);
```

Comment:
Doesn't dequeue release the message?

And, now that I'm looking at this, do we even want this line? It looks like it is trying to feed packets we sent out back into the stack.